Library/Homebrew/dev-cmd/boneyard-formula-pr.rb

```diff
@@ -0,0 +1,135 @@
+# Creates a pull request to boneyard a formula.
+#
+# Usage: brew boneyard-formula-pr [options...] <formula-name>
+#
+# Options:
+#   --dry-run:  Print what would be done rather than doing it.
+
+require "formula"
+require "utils/json"
+require "fileutils"
+
+begin
+  require "json"
+rescue LoadError
+  puts "Homebrew does not provide Ruby dependencies; install with:"
+  puts "  gem install json"
+  odie "Dependency json is not installed."
+end
+
+module Homebrew
+  def boneyard_formula_pr
+    formula = ARGV.formulae.first
+    odie "No formula found!" unless formula
+
+    formula_path = formula.path
+    formula_file = "#{formula.name}.rb"
+    bottle_block = File.read(formula_path).include? "  bottle do"
+    boneyard_tap = Tap.fetch("homebrew", "boneyard")
+    tap_migrations_path = formula.tap.path/"tap_migrations.json"
+    if ARGV.dry_run?
+      puts "brew update"
+      puts "brew tap #{boneyard_tap.name}"
+      puts "cd #{formula.tap.path}"
+      cd formula.tap.path
+      puts "cp #{formula_path.relative_path_from(formula.tap.path)} #{boneyard_tap.formula_dir}"
```

Comment:
I think you're making things too complicated. `boneyard_tap.path` returns the path you're trying to construct here (even if not tapped), it's just `Tap#formula_dir` that requires the tap to be physically present. You can adjust this once formulae in `homebrew/boneyard` actually move to `Formula/`.

> This works `formula.tap.path.parent/boneyard_tap.name.sub("/","-")`

Not a concern for our use case, but this only works because the prefix `homebrew-` happens to match `Tap#user` here. For something like `uniqmartin/tools` it would be totally wrong and print `…/Taps/homebrew/uniqmartin-tools` instead of `…/Taps/uniqmartin/homebrew-tools`.